modules/editorial-metadata/editorial-metadata.php

```diff
@@ -286,66 +251,114 @@ public static function get_editorial_metadata_term_by( $field, $value ) {
+	 * @return array $terms The terms with metadata added
+	 */
+	public static function add_metadata_to_terms( array $terms, array $taxonomies ): array {
+		if ( ! in_array( self::METADATA_TAXONOMY, $taxonomies ) ) {
+			return $terms;
+		}
+
+		foreach ( $terms as $term ) {
+			$term = self::add_metadata_to_term( $term );
+		}
+
+		return $terms;
 	}
 
 	/**
 	 * Insert a new editorial metadata term
-	 * @todo Handle conflicts with existing terms at that position (if relevant)
+	 *
+	 * @param array $args The arguments for the new term
+	 * @return WP_Term|WP_Error $term The new term or a WP_Error object if something disastrous happened
 	 */
 	public static function insert_editorial_metadata_term( array $args ): WP_Term|WP_Error {
-		// Term is always added to the end of the list
-		$default_position = count( self::get_editorial_metadata_terms() ) + 1;
-
-		$defaults  = [
-			'position'    => $default_position,
-			'name'        => '',
-			'slug'        => '',
-			'description' => '',
-			'type'        => '',
-		];
-		$args      = array_merge( $defaults, $args );
-		$term_name = $args['name'];
-		unset( $args['name'] );
-
-		// We're encoding metadata that isn't supported by default in the term's description field
-		$args_to_encode = [
-			'description' => $args['description'],
-			'position'    => $args['position'],
-			'type'        => $args['type'],
+		$term_to_save = [
+			'slug'        => $args['slug'] ?? '',
+			'description' => $args['description'] ?? '',
 		];
+		$term_name = $args['name'] ?? '';
 
-		$encoded_description = TaxonomyUtilities::get_encoded_description( $args_to_encode );
-		$args['description'] = $encoded_description;
-
-		unset( $args['position'] );
-		unset( $args['type'] );
+		$inserted_term = wp_insert_term( $term_name, self::METADATA_TAXONOMY, $term_to_save );
 
-		$inserted_term = wp_insert_term( $term_name, self::METADATA_TAXONOMY, $args );
+		if ( is_wp_error( $inserted_term ) ) {
+			return $inserted_term;
+		}
 
 		// Reset the internal object cache
 		self::$editorial_metadata_terms_cache = [];
 
-		// Populate the inserted term with the new values, or else only the term_taxonomy_id and term_id are returned.
-		if ( is_wp_error( $inserted_term ) ) {
-			return $inserted_term;
-		} else {
-			// Update the term with the meta_key, as we use the term_id to generate it
-			self::update_editorial_metadata_term( $inserted_term['term_id'] );
-			$inserted_term = self::get_editorial_metadata_term_by( 'id', $inserted_term['term_id'] );
+		$term_id = $inserted_term['term_id'];
+
+		$metadata_type = $args['type'] ?? '';
+		$metadata_postmeta_key  = self::get_postmeta_key( $metadata_type, $term_id );
+
+		$type_meta_result = add_term_meta( $term_id, self::METADATA_TYPE_KEY, $metadata_type );
+		if ( is_wp_error( $type_meta_result ) ) {
+			return $type_meta_result;
+		} else if ( ! $type_meta_result ) {
+			// If we can't save the type, we should delete the term
+			wp_delete_term( $term_id, self::METADATA_TAXONOMY );
```

Comment:
Added in some cleanup code in case of a failure. My whole site failed when this happened so this is critical imo